xl/player/gst/engine.py

```diff
@@ -413,10 +413,22 @@ def reconfigure_sink(self):
 
         self.audio_sink.reconfigure(sink)
 
-        # We also need to reset playbin
-        old_state = self.get_gst_state()
-        self.playbin.set_state(Gst.State.NULL)
-        self.playbin.set_state(old_state)
+        # We also need to restart playback if playing or paused
+        state = self.get_gst_state()
+        if state == Gst.State.PLAYING or state == Gst.State.PAUSED:
+            track = self.current_track
+            position = self.get_position() / 1e9
+            if state == Gst.State.PLAYING:
+                position = position - 1
+            volume = self.get_volume()
+            self.set_volume(0)
+            GLib.timeout_add_seconds(
+                1, self._restart, track, position, state == Gst.State.PAUSED, volume
```

Comment:
This is weird, why is there a whole second delay here? This seems ripe for causing race conditions.

Reply:
The problem is to deal with the timeout. 
In this way I can seek to one second before.

Reply:
I don't claim to understand the problem yet, but this doesn't feel like a good way to solve it.

Reply:
Problem is described here: https://github.com/exaile/exaile/issues/801#issuecomment-1026314414

May be this should be solved in the dynamic_sink?